### ISSUE TITLE
[DOC] Loader sequence does matter, especially when using uglifyJSPlugin

### ISSUE THREAD

**Issue by coodoo** [webpack/webpack#888]
@sokra Might worth adding this to the doc somewhere to save people's time in the future :)

**tl;dr:** if you use both `babeljs` and `react-hot` loader, put `react-hot` loader before `babeljs`, otherwise when uglifying the source you will run into errors.

Here's the correct way to do it:

```js
module: {
    
    loaders: [{
      test: /\.jsx?$/,
      exclude: [ /node_modules/, /app\/vendor/ ],
      loaders: [ 'react-hot' , 'babel-loader']
    }]
  },
```

Errors you might run into looks like this:
`
RROR in ./app/js/boot.js

Module build failed: TypeError: /Users/foo/bar/node_modules/react-hot-loader/index.js!/Users/foo/bar/app/js/boot.js: Cannot read property 'match' of undefined

    at urlParse (/Users/foo/bar/node_modules/babel-core/node_modules/source-map/lib/source-map/util.js:37:21)

    at normalize (/Users/foo/bar/node_modules/babel-core/node_modules/source-map/lib/source-map/util.js:86:15)
`

**Comment by sebastienbarre** [comment]
Thanks a lot @coodoo, I had been googling for this "```Cannot read property 'match' of undefined```" error for a while now, until your post popped up. My issue here is that I was putting ```react-hot``` *in front* of ```babel``` but inside the ```loader``` property, as in ```loader: "react-hot!babel"```; this does not work, you have to use the ```loaders``` property instead, as in ```loaders: ["react-hot", "babel"]```. 
Thanks for the tip.

**Comment by bdefore** [comment]
I also had this issue where the loader sequence mattered. Only the output `Cannot read property 'match' of undefined` leading me here led me to the solution.

**Comment by pksjce** [comment]
This issue was moved to webpack/webpack.io#142
